src/software/backend/simulation/simulator_robot_singleton.h

```diff
@@ -0,0 +1,232 @@
+#pragma once
+
+#include <cinttypes>
+#include <optional>
+
+#include "software/backend/simulation/simulator_robot.h"
+extern "C"
+{
+#include "firmware/main/app/world/firmware_robot.h"
+}
+
+// TODO: These are all hardcoded values copied from firmware/main/physics/physics.h
+// and firmware/main/control/control.h
+// They should be replaced with the proper constants once firmware cleanup is done
+#define GEAR_RATIO 0.5143f  // define as speed multiplication from motor to wheel
+#define WHEEL_RADIUS 0.0254f
+#define WHEEL_SLIP_VOLTAGE_LIMIT 4.25f  // Voltage where wheel slips (acceleration cap)
+#define RPM_TO_VOLT (1.0f / 374.0f)     // motor RPM to back EMF
+#define PHASE_RESISTANCE 1.6f           // adjust this number as calculated
+#define CURRENT_PER_TORQUE 39.21f       // from motor data sheet (1/25.5 mNm)
+#define ROBOT_POINT_MASS 2.48f
+#define ROBOT_RADIUS 0.085f
+#define INERTIAL_FACTOR 0.37f
+#define ROT_MASS (INERTIAL_FACTOR * ROBOT_POINT_MASS)
+#define INERTIA (ROT_MASS * ROBOT_RADIUS * ROBOT_RADIUS)
+#define JERK_LIMIT 40.0f  //(m/s^3)
+
+/**
+ * Because the FirmwareRobot_t struct is defined in the .c file (rather than the .h file),
+ * C++ considers it an incomplete type and is unable to use it with smart pointers
+ * because it doesn't know the size of the object. Therefore we need to create our own
+ * "Deleter" class we can provide to the smart pointers to handle that instead.
+ *
+ * See https://en.cppreference.com/w/cpp/memory/unique_ptr/unique_ptr for more info and
+ * examples
+ */
+struct FirmwareRobotDeleter
+{
+    void operator()(FirmwareRobot_t* firmware_robot) const
+    {
+        app_firmware_robot_destroy(firmware_robot);
```

Comment:
We should destroy the members first as well.

Reply:
Ah good catch. I started trying this out, but quickly realized I can't destroy the child members unless I also define a "destroyer" for them. I don't really want to have to do this for everything if we're planning on having the object responsible for de-allocating its children soon anyway, so is it fine if I leave this (perhaps with a comment like in the other areas)?

Reply:
Let's change it to something like this for now:
```
Wheel_t* front_right_wheel = app_firmware_robot_getFrontRightWheel(firmware_robot);
...
Chicker_t* chicker = app_firmware_robot_getChicker(firmware_robot);
...
app_firmware_robot_destroy(firmware_robot);
app_wheel_destroy(front_right_wheel);
...
and leave a link to the issue, rather then just leaking memory. See here for what we need to delete: https://github.com/UBC-Thunderbots/Software/blob/master/src/firmware/main/app/world/firmware_robot.c#L24

Reply:
didn't think of that. done.

Reply:
Doesn't look like this got changed?

Reply:
pushed, now should show up